tests/tensor/rewriting/test_math.py

```diff
@@ -2031,6 +2031,47 @@ def test_exp_log_nested(self, nested_expression, expected_switches):
         assert len(ops_graph) == expected_switches
 
 
+class TestSqrSqrt:
+    def setup_method(self):
+        mode = get_default_mode()
+        self.mode = mode.including(
+            "local_sqrt_sqr",
+        ).excluding("fusion")
+        self.rng = np.random.default_rng()
+
+    def test_sqr_sqrt(self):
+        # sqrt(x) ** 2 -> x
+        x = pt.tensor("x", shape=(None, None))
+        out = sqr(sqrt(x))
+        out = rewrite_graph(out, include=["canonicalize", "specialize", "stabilize"])
+
+        assert equal_computations([out], [pt_abs(x)])
+
+    def test_sqrt_sqr(self):
+        x = pt.tensor("x", shape=(None, None))
+        out = sqrt(sqr(x))
+        out = rewrite_graph(out, include=["canonicalize", "specialize", "stabilize"])
+
+        expected = switch(
+            ge(x, np.zeros((1, 1), dtype="int8")),
+            x,
+            np.full((1, 1), np.nan, dtype=x.type.dtype),
+        )
+
+        assert equal_computations([out], [expected])
+
+    def test_sqr_sqrt_integer_upcast(self):
+        x = ivector("x")
+        f = function([x], sqrt(sqr(x)), mode=self.mode)
+        nodes = f.maker.fgraph.toposort()
+        assert not any(
+            isinstance(node.op, Elemwise)
+            and isinstance(node.op.scalar_op, ps.Sqr | ps.Sqrt)
+            for node in nodes
+        )
+        assert f([1, 2, 3]).dtype in ["float32", "float64"]
```

Comment:
Nit, if you already bother to compile and eval why not check with the expected output? strict=True in np.testing.asser_array_equal enforces dtypes.

But I wouldn't compile either. This check you had is equivalent to asserting the dtype of the output of the rewrite. And/or you can use the equal_computations approach. Approving already anyway, up to you.